src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/cursor/CursorCloseExecutor.java

```diff
@@ -0,0 +1,103 @@
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.executor.cursor;
+
+import com.amazon.opendistroforelasticsearch.sql.cursor.CursorType;
+import com.amazon.opendistroforelasticsearch.sql.cursor.DefaultCursor;
+import com.amazon.opendistroforelasticsearch.sql.metrics.MetricName;
+import com.amazon.opendistroforelasticsearch.sql.metrics.Metrics;
+import com.amazon.opendistroforelasticsearch.sql.rewriter.matchtoterm.VerificationException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.search.ClearScrollResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestChannel;
+import org.json.JSONException;
+
+import java.util.Map;
+
+import static org.elasticsearch.rest.RestStatus.OK;
+
+public class CursorCloseExecutor implements CursorRestExecutor {
+
+    private static final Logger LOG = LogManager.getLogger(CursorCloseExecutor.class);
+
+    private static final String SUCCEEDED_TRUE = "{\"succeeded\":true}";
+    private static final String SUCCEEDED_FALSE = "{\"succeeded\":false}";
+
+    private String cursorId;
+
+    public CursorCloseExecutor(String cursorId) {
+        this.cursorId = cursorId;
+    }
+
+    public void execute(Client client, Map<String, String> params, RestChannel channel) throws Exception {
+        try {
+            String formattedResponse = execute(client, params);
+            channel.sendResponse(new BytesRestResponse(OK, "application/json; charset=UTF-8", formattedResponse));
+        } catch (IllegalArgumentException | JSONException e) {
+            Metrics.getInstance().getNumericalMetric(MetricName.FAILED_REQ_COUNT_CUS).increment();
+            LOG.error("Error parsing the cursor", e);
+            channel.sendResponse(new BytesRestResponse(channel, e));
+        } catch (ElasticsearchException e) {
+            int status = (e.status().getStatus());
+            if (status > 399 && status < 500) {
+                Metrics.getInstance().getNumericalMetric(MetricName.FAILED_REQ_COUNT_CUS).increment();
+            } else if (status > 499) {
+                Metrics.getInstance().getNumericalMetric(MetricName.FAILED_REQ_COUNT_SYS).increment();
+            }
+            LOG.error("Error completing cursor request", e);
+            channel.sendResponse(new BytesRestResponse(channel, e));
+        }
+    }
+
+    public String execute(Client client, Map<String, String> params) throws Exception {
```

Comment:
Two concerns.
1. The execute method is almost as same as CursorResultExecutor.execute, only difference is the LOG.error. Is it possible to avoid the duplication.
2. In case we already define the Curser, could we construct CursorCloseExecutor by using cursor and encapsulate the Cursor construct operation in CursorFactory?

Reply:
Sure, will give a shot to refactor as described.

Reply:
Keeping it as it now, little tricky to refactor.

Reply:
In my opinion, the cursor format ":" which should be encapsulated to Cursor implementation and should not been exposed to caller.
We can leave it for now for time consideration, Please consider add UT to cover the logic in CursorCloseExecutor and CursorResultExecutor. If possible, add the issue to tracking the refactoring.

Reply:
https://github.com/opendistro-for-elasticsearch/sql/issues/426.

Adding UTs.